clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -3824,19 +3824,13 @@ protected Set<String> maybeSendIngestionHeartbeat() {
       // Not time for another heartbeat yet.
       return null;
     }
-    Set<String> failedPartitions = sendHeartbeatToRealtimeTopic(false);
-    if (isSeparatedRealtimeTopicEnabled()) {
-      failedPartitions.addAll(sendHeartbeatToRealtimeTopic(true));
-    }
-    lastSendIngestionHeartbeatTimestamp.set(currentTimestamp);
-    return failedPartitions;
-  }
 
-  Set<String> sendHeartbeatToRealtimeTopic(boolean isSeparateTopic) {
-    AtomicInteger numHeartBeatSuccess = new AtomicInteger(0);
     List<CompletableFuture<PubSubProduceResult>> heartBeatFutures = new ArrayList<>();
+    List<CompletableFuture<PubSubProduceResult>> heartBeatFuturesForSepRT = new ArrayList<>();
     Set<String> failedPartitions = VeniceConcurrentHashMap.newKeySet();
+    Set<String> failedPartitionsForSepRT = VeniceConcurrentHashMap.newKeySet();
     AtomicReference<CompletionException> completionException = new AtomicReference<>(null);
```

Comment:
Can we init collections size == `pcs.values().size()` and  sepRT collections empty if the feature is not enabled?